src/shogun/kernel/GaussianARDKernel.cpp

```diff
@@ -122,21 +107,10 @@ SGMatrix<float64_t> CGaussianARDKernel::get_parameter_gradient(
 		{
 			for (index_t k=0; k<num_rhs; k++)
 			{
-				SGVector<float64_t> avec=
-					((CDenseFeatures<float64_t>*) lhs)->get_feature_vector(j);
-				SGVector<float64_t> bvec=
-					((CDenseFeatures<float64_t>*) rhs)->get_feature_vector(k);
-
-				REQUIRE(avec.vlen==bvec.vlen, "Number of right and left hand "
-						"features must be the same\n");
-
-				float64_t result=0;
+				float64_t result=distance(j,k);
```

Comment:
Yes. It could be done. 
The trick part is gradient wrt to parameters (eg, weights). ( in fact, in the future, automatic differentiation may be a good choice for gradient stuff when hand-crafted gradients are not available)    
Usually, the full pairwise matrix should be positive definitive.   (Not sure whether the linear algebra back end supports such checking since Eigen cannot be used in core API)
I am working on it. ( will soon be available next week). ( I am looking at codes about the `linalg`)